### PR TITLE
Added Funnel component storybook. (Issue #3430)

### PR DESCRIPTION
**Pull request by d-gottlieb**
## Description

Added Storybook documentation for the `Funnel` component.

## Related Issue

[Issue #3430](https://github.com/recharts/recharts/issues/3430)

## Motivation and Context

This adds more comprehensive documentation to Storybook for recharts in the effort to move off of API docs.

## How Has This Been Tested?

I compared the output of Storybook to the APi documentation that already exists.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: storybook/stories/API/props/CartesianComponentShared.ts [hunk 1]
```diff
@@ -9,6 +9,34 @@ export const dataKey = {
...
+    category: 'General',
+  },
+};
+
+export const trapezoids = {
+  description: 'The coordinates of all trapezoids in the chart, usually calculated internally',
+  table: {
+    type: {
+      summary: 'Array',
+      detail: '[{x: 12, y: 12, upperWidth: 240, lowerWidth: 22, height: 80,}]',
```
Comment (ckifer) [inline storybook/stories/API/props/CartesianComponentShared.ts:35]: nit nit nit, non-blocking: trailing comma after height

File: storybook/stories/API/props/ChartProps.ts [hunk 2]
```diff
@@ -45,6 +45,16 @@ export const margin = {
   },
 };
 
+export const dataKey = {
+  description: 'The key or getter of a gbroup of data which should be unique in a LineChart',
```
Comment (ckifer) [inline storybook/stories/API/props/ChartProps.ts:49]: typo: `gbroup` -> `group`

Also, where is this used? It looks like right now its specific to a `LineChart`?

**Review by ckifer — COMMENTED** [review]
few nits

File: storybook/stories/API/props/ChartProps.ts (hunk 2 above)
Comment (d-gottlieb) [reply storybook/stories/API/props/ChartProps.ts:49]: > typo: `gbroup` -> `group`
> 
> Also, where is this used? It looks like right now its specific to a `LineChart`?

@ckifer This is showing in [the Funnel API docs](https://recharts.org/en-US/api/Funnel). Is this not right? Should it be removed?

File: storybook/stories/API/props/CartesianComponentShared.ts (hunk 1 above)
Comment (d-gottlieb) [reply storybook/stories/API/props/CartesianComponentShared.ts:35]: Fixed!

File: storybook/stories/API/props/ChartProps.ts (hunk 2 above)
Comment (ckifer) [reply storybook/stories/API/props/ChartProps.ts:49]: It definitely shouldn't be `LineChart`, so I think a typo in the old docs. I would say replace with `FunnelChart`

if you look at [`Line` ](https://recharts.org/en-US/api/Line)its identical, so I think a copy-pasta

Comment (d-gottlieb) [reply storybook/stories/API/props/ChartProps.ts:49]: @ckifer Fixed!

**Review by ckifer — APPROVED** [review]
🚀

File: storybook/stories/API/props/ChartProps.ts [hunk 3]
```diff
@@ -45,6 +45,16 @@ export const margin = {
   },
 };
 
+export const dataKey = {
+  description: 'The key or getter of a gbroup of data which should be unique in a FunnelChart',
```
Comment (ckifer) [inline storybook/stories/API/props/ChartProps.ts:49]: Actually, I'm still confused. This is in the common file, but is now specific to `FunnelChart` - it also isn't imported and used in the storybook above. Maybe we can just remove the specification of a chart type and call it common?

Still have typo `gbroup`

Comment (d-gottlieb) [reply storybook/stories/API/props/ChartProps.ts:49]: > Actually, I'm still confused. This is in the common file, but is now specific to `FunnelChart` - it also isn't imported and used in the storybook above. Maybe we can just remove the specification of a chart type and call it common?
> 
> Still have typo `gbroup`

@ckifer Yeah, that makes sense. I changed the description and fixed the typo.

**Review by ckifer — APPROVED** [review]

**Pull request merged**